src/rpm/package.rs

```diff
@@ -1019,3 +1137,53 @@ impl PackageMetadata {
         }
     }
 }
+
+pub struct FileIterator<'a> {
+    file_entries: Vec<FileEntry>,
+    archive: Box<dyn io::Read + 'a>,
+    count: usize,
+}
+
+#[derive(Debug)]
+pub struct RpmFile {
+    pub metadata: FileEntry,
+    pub content: Vec<u8>,
```

Comment:
Can we make this a slice or an offset regarding the actual package reader? I don't think we should read this to memory at all.